app/src/behaviors/behavior_turbo_key.c

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_turbo_key
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <zmk/hid.h>
+#include <zmk/event_manager.h>
+#include <zmk/events/keycode_state_changed.h>
+#include <zmk/behavior.h>
+#include <zmk/behavior_queue.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct behavior_turbo_config {
+    int tap_ms;
+    int wait_ms;
+    int toggle_term_ms;
+    const struct zmk_behavior_binding binding;
+};
+
+#define ZMK_BHV_TURBO_MAX_ACTIVE 5
+
+struct active_turbo {
+    const struct behavior_turbo_config *config;
+    uint32_t position;
+    bool is_active;
+    bool is_pressed;
+
+    int32_t press_time;
+
+    // Timer Data
+    bool timer_started;
+    bool timer_cancelled;
+    bool turbo_decided;
+    int64_t release_at;
+    struct k_work_delayable release_timer;
+};
+
+struct active_turbo active_turbos[ZMK_BHV_TURBO_MAX_ACTIVE] = {};
```

Comment:
All of this seems a bit overly complex for this... Why not just have a single delayable work that lives in the "data" struct tied to each instance of the behavior?

That would avoid all the tracking of active turbos, etc.

Seems this might have been inspired by the hold-tap logic... I'm not sure this has the same needs/complexity as that.

Reply:
Fixed in recent commits.